core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala

```diff
@@ -152,7 +152,7 @@ class RadixSortSuite extends SparkFunSuite with Logging {
       val (buf1, buf2) = generateKeyPrefixTestData(N, rand.nextLong & 0xff)
       referenceKeyPrefixSort(buf1, 0, N, sortType.referenceComparator)
       val outOffset = RadixSort.sortKeyPrefixArray(
-        buf2, N, sortType.startByteIdx, sortType.endByteIdx,
+        buf2, 0, N, sortType.startByteIdx, sortType.endByteIdx,
```

Comment:
Also, it'd be great to explicitly test boundary case for an array with all nulls